tests/lexer/tstrnumlits.nim

```diff
@@ -123,5 +116,34 @@ template main =
     doAssert 0.0e-12'wrap == "[[0.0e-12]]"
     doAssert 0e-12'wrap == "[[0e-12]]"
 
+  block: # macro and template usage
+    template `'foo`(a: string): untyped = (a, 2)
+    doAssert -12'foo == ("-12", 2)
+    template `'fooplus`(a: string, b: int): untyped = (a, b)
+    doAssert -12'fooplus(2) == ("-12", 2)
+    template `'fooplusopt`(a: string, b: int = 99): untyped = (a, b)
+    doAssert -12'fooplusopt(2) == ("-12", 2)
+    doAssert -12'fooplusopt() == ("-12", 99)
+    doAssert -12'fooplusopt == ("-12", 99)
+    macro `'bar`(a: static string): untyped = 
+      var infix = newNimNode(nnkInfix)
+      infix.add newIdentNode("&")
+      infix.add newLit("got ")
+      infix.add newLit(a.repr)
+      result = newNimNode(nnkStmtList)
+      result.add infix
+    doAssert -12'bar == "got \"-12\""
+    macro deb(a): untyped = newLit(a.repr)
+    doAssert deb(-12'bar) == "-12'bar"
+    # macro metawrap(): untyped =
+    #   func wrap1(a: string): string = "{" & a & "}"
+    #   func `'wrap2`(a: string): string = "{" & a & "}"
+    #   result = quote do:
+    #     let a1 = wrap1"-128"
+    #     let a2 = -128'wrap2
+    # metawrap()
+    # doAssert a1 == "{-128}"
+    # doAssert a2 == "{-128}"
```

Comment:
```suggestion
    macro `'bar`(a: static string): untyped = newLit(a.repr)
    doAssert -12'bar == "\"-12\""
    macro deb(a): untyped = newLit(a.repr)
    doAssert deb(-12'bar) == "-12'bar"
    when false: # xxx doesn't work yet
      macro metawrap(): untyped =
        func wrap1(a: string): string = "{" & a & "}"
        func `'wrap2`(a: string): string = "{" & a & "}"
        result = quote do:
          let a1 = wrap1"-128"
          let a2 = -128'wrap2
      metawrap()
      doAssert a1 == "{-128}"
      doAssert a2 == "{-128}"
```

* when false is better than commenting out when it parses ok (as here)
* the `nnkInfix` didn't add value to this test IMO (unrelated to litterals)